src/MagnumPlugins/UfbxImporter/UfbxImporter.h

```diff
@@ -53,6 +53,241 @@
+    @cb{.ini} preserveRootNode @ce @ref Trade-UfbxImporter-configuration "configuration option" is enabled.
+-   FBX files may contain nodes with "geometric transforms" that transform only
+    the mesh of the node without affecting children. These are converted to
+    unnamed helper nodes in the scene unless the @cb{.ini} geometricTransformNodes @ce
+    @ref Trade-UfbxImporter-configuration "configuration option" is explicitly disabled.
+-   Imported scenes always have @ref SceneMappingType::UnsignedInt, with
+    @ref SceneData::mappingBound() equal to @ref objectCount(). The scene is
+    always 3D.
+-   All reported objects have a @ref SceneField::Parent (of type
+    @ref SceneFieldType::Int), @ref SceneField::Translation (of type @ref SceneFieldType::Vector3d),
+    @ref SceneField::Rotation (of type @ref SceneFieldType::Quaterniond),
+    @ref SceneField::Scaling (of type @ref SceneFieldType::Vector3d) and
+    importer-specific @cpp "Visibility" @ce (of type @ref SceneFieldType::UnsignedByte
+    representing a boolean value).
+    These five fields share the same object mapping with
+    @ref SceneFieldFlag::ImplicitMapping set.
+-   If the scene references meshes, a @ref SceneField::Mesh (of type
+    @ref SceneFieldType::UnsignedInt) and a @ref SceneField::MeshMaterial (of
+    type @ref SceneFieldType::Int) is present, both with
+    @ref SceneFieldFlag::OrderedMapping set. Missing material IDs are @cpp -1 @ce.
+    If a mesh contains multiple materials it is split into parts and the node
+    contains each part as a separte mesh/material entry.
+    The same mesh can appear instanced multiple times under many
+    nodes with different materials. See @ref Trade-UfbxImporter-instancing for further details.
+-   If the scene references cameras, a @ref SceneField::Camera (of type
+    @ref SceneFieldType::UnsignedInt) is present, with
+    @ref SceneFieldFlag::OrderedMapping set. A single camera can be referenced
+    by multiple nodes. See @ref Trade-UfbxImporter-instancing for further
+    details.
+-   If the scene references lights, a @ref SceneField::Light (of type
+    @ref SceneFieldType::UnsignedInt) is present, with
+    @ref SceneFieldFlag::OrderedMapping set. A single light can be referenced
+    by multiple nodes. See @ref Trade-UfbxImporter-instancing for further
+    details.
+
+@subsection Trade-UfbxImporter-behavior-materials Material import
+
+-   Supports both legacy FBX Phong material model and more modern PBR
+    materials, in some cases both are defined as PBR materials may have
+    a legacy Phong material filled as a fallback.
+-   The legacy FBX material model and most PBR material models have factors
+    for various attributes, by default these are premultiplied into the value
+    but you can retain them using the @cb{.ini} preserveMaterialFactors @ce @ref Trade-UfbxImporter-configuration "configuration option"
+-   Two-sided property and alpha mode is not imported.
```

Comment:
It's copypasted but with *intention* :D There is `UFBX_MATERIAL_FEATURE_DOUBLE_SIDED` but I'm considering deleting that actually now that you reminded me of it, as I can see that it can be a bit confusing. The only place that enables that is gLTF-compatible material in 3ds Max, so it's *extremely* rare. Though now that you mention it I should probably look into if FBX has other ways of representing that, as it is a pretty basic feature.

Reply:
Hm, so should all materials be considered as double-sided _and alpha-blended_ by default? I.e., is that how FBX treats them?

Just few days ago I was considering writing a tool to discover if a mesh is manifold, because I have a certain glTF dataset which has the double-sided material property set implicitly for *everything*, which is "very nice" runtime-perf-wise. Some of the models are actual cloths etc. which need it, but majority of them doesn't.

In any case I'd reword the comment slightly to not sound like "FBX has the properties but we decided to ignore them" -- though I still need to get a clarification on how to deal with alpha blending without having to go through all texture data and looking for `alpha < 1.0` :sweat_smile: Implicitly treating everything as alpha-blended seems extremely expensive, compared to that implicitly treating everything as double-sided would still be fine.

```suggestion
-   FBX materials don't have any equivalent for the 
    @ref MaterialAttribute::DoubleSided and 
    @ref MaterialAttribute::AlphaMask properties. Alpha blending is always 
    (???) enabled.
```

Or is this one of those features that make a difference between an "asset delivery" and "asset exchange" format, in that the loading application is expected to figure out the rendering efficiency properties on its own?

Reply:
Looked around a bit and seems like FBX doesn't really seem to have any indication whether surfaces are double sided. I wouldn't set `MaterialAttribute::DoubleSided` by default myself unconditionally as the file doesn't really have any opinion about it and many single sided surfaces are exported in the same manner as double-sided ones.

Alpha is tricky for sure as by convention materials might have textures with an alpha channel for transparency, but this is not indicated in any way in the file. We could do some detection whether the material has a mask texture in the opacity field or something, but those are going to be a large minority in real-time assets.

And I don't think it really has to do much with being an "asset exchange" format, just more that the material model in FBX is sorely underspecified and vendor-specific. Outside of some very esoteric 3ds Max materials we just have no idea what the `DoubleSided/AlphaMask` should be.

I don't think we should be (at least by default) overly conservative with double-sided/alpha as then we risk users taking a potentially unobservable performance hit. One option would be if `MaterialData` would have a way of signaling that the `DoubleSided` and `AlphaMask/Blend` attributes are unknown, as that would accurately represent the file, but we can also just document it to the behavior section of the importer.

Reply:
Thanks for the investigation -- and I see you already updated the docs, so that's great.

> We could do some detection whether the material has a mask texture in the opacity field or something

I'll have to write tools to undo unnecessary alpha blending / double sided bits for asset cleanup anyway, and adding a related tool to do the inverse would be trivial. But that operation would be inherently slow (because it'd be opening all referenced textures and calculating a minimum value of their alpha channel), so it's not something to do implicitly.